### DISCUSSION TITLE
[odb, dbDatebase, getTechs] How to iterate over dbSet items returned by getTechs() of class dbDatabase

### DISCUSSION THREAD

**Discussion by zpzhu99** [The-OpenROAD-Project/OpenROAD#6207] [Q&A] [answered]
Hi, community,

I'm working with `obd` on my current project and would appreciate any help. In the following demo, I read a LEF file in as a `dbDatabase` and tried to extract all the techs with `getTechs`(). I found that the returned item is a swig object. Can anyone please let me know how to iterate over this and save each item here as `dbTech`? 
Meanwhile, I noticed that in `OpenROAD/src/odb/src/swig/python/dbtypes.i`, a `dbSet` should have been converted to a Python list but it is not here. I would be grateful if someone could try to reproduce my case so that I can know if the problem is caused by building the OpenROAD or something else. Thanks!

```
[xx@xx demo]$ openroad -python
OpenROAD v2.0-16761-g688a7722f 
Features included (+) or not (-): +Charts +GPU +GUI +Python
This program is licensed under the BSD-3 license. See the LICENSE file for details.
Components of this program may be licensed under more restrictive licenses which must be honored.
[WARNING ORD-0039] .openroad ignored with -python
Python 3.6.8 (default, Apr 25 2024, 09:54:46) 
[GCC 8.5.0 20210514 (Red Hat 8.5.0-22)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import openroad
>>> import odb
>>> import os
>>> db = odb.dbDatabase.create()
>>> odb.read_lef(db, './sky130_fd_sc_hd.tlef')
[INFO ODB-0227] LEF file: ./sky130_fd_sc_hd.tlef, created 13 layers, 25 vias
<odb.dbLib; proxy of <Swig Object of type 'odb::dbLib *' at 0x7f22ed7f67e0> >
>>> techs = db.getTechs()
>>> techs
<Swig Object of type 'dbSet< odb::dbTech > *' at 0x7f22ed81a7b0>
>>> 
```

**Comment by rovinski** [comment]
@maliberty this is related to a project we are working on. Is this expected behavior, or maybe you can point us to how the interface file needs to be changed so that we can make a PR?

**Reply by maliberty** [reply]
It is not expected and thought https://github.com/The-OpenROAD-Project/OpenROAD/blob/17a6ee54857ea8ea342dc37b9e15ace3426b3121/src/odb/src/swig/python/dbtypes.i#L29 would have covered this.

**Reply by maliberty** [reply]
I would also suggest you use the Tech API though I don't think it will solve this particular issue.

**Reply by maliberty** [reply]
I see the issue is common/containers.i doesn't cover dbTech - PR coming.

**Reply by rovinski** [reply, marked as answer]
Fixed in #6217.
